File: java/src/main/java/org/example/schema/KnowSchema.java
package org.example.schema;

import io.milvus.v2.client.ConnectConfig;
import io.milvus.v2.client.MilvusClientV2;
import io.milvus.v2.service.collection.request.CreateCollectionReq;

import io.milvus.v2.common.DataType;
import io.milvus.v2.service.collection.request.AddFieldReq;

public class KnowSchema {
    private static final MilvusClientV2 client;
    static {
        client = new MilvusClientV2(ConnectConfig.builder()
                .uri("http://localhost:19530")
                .build());
    }

    public static void main(String[] args) {
        CreateCollectionReq.CollectionSchema schema = client.createSchema();

        schema.addField(AddFieldReq.builder()
                .fieldName("my_id")
                .dataType(DataType.Int64)
                // highlight-start
                .isPrimaryKey(true)
                .autoID(false)
                // highlight-end
                .build());

        schema.addField(AddFieldReq.builder()
                .fieldName("my_vector")
                .dataType(DataType.FloatVector)
                // highlight-next-line
                .dimension(5)
                .build());

        schema.addField(AddFieldReq.builder()
                .fieldName("my_varchar")
                .dataType(DataType.VarChar)
                // highlight-next-line
                .maxLength(512)
                .build());

        schema.addField(AddFieldReq.builder()
                .fieldName("my_int64")
                .dataType(DataType.Int64)
                .build());

        schema.addField(AddFieldReq.builder()
                .fieldName("my_bool")
                .dataType(DataType.Bool)
                .build());

        schema.addField(AddFieldReq.builder()
                .fieldName("my_json")
                .dataType(DataType.JSON)
                .build());

        schema.addField(AddFieldReq.builder()
                .fieldName("my_array")
                .dataType(DataType.Array)
                .elementType(DataType.VarChar)
                .maxCapacity(5)
                .maxLength(512)
                .build());

    }
}

File: python/schema/know_schema.py
from pymilvus import MilvusClient, DataType

schema = MilvusClient.create_schema()

schema.add_field(
    field_name="my_id",
    datatype=DataType.INT64,
    # highlight-start
    is_primary=True,
    auto_id=False,
    # highlight-end
)

schema.add_field(
    field_name="my_vector",
    datatype=DataType.FLOAT_VECTOR,
    # highlight-next-line
    dim=5
)

schema.add_field(
    field_name="my_varchar",
    datatype=DataType.VARCHAR,
    # highlight-next-line
    max_length=512
)

schema.add_field(
    field_name="my_int64",
    datatype=DataType.INT64,
)

schema.add_field(
    field_name="my_bool",
    datatype=DataType.BOOL,
)

schema.add_field(
    field_name="my_json",
    datatype=DataType.JSON,
)

schema.add_field(
    field_name="my_array",
    datatype=DataType.ARRAY,
    element_type=DataType.VARCHAR,
    max_capacity=5,
    max_length=512,
)